creator-node/src/utils.js

```diff
@@ -196,6 +202,84 @@ const ipfsGet = (path, req, timeout = 1000) => new Promise(async (resolve, rejec
   }
 })
 
+async function findCIDInNetwork (filePath, cid, logger) {
+  const attemptedStateFix = await getIfAttemptedStateFix(filePath)
+  if (attemptedStateFix) return
+
+  // generate signature
+  const delegateWallet = config.get('delegateOwnerWallet').toLowerCase()
+  const { signature, timestamp } = generateTimestampAndSignature({ filePath, delegateWallet }, config.get('delegatePrivateKey'))
+  const creatorNodes = await getAllRegisteredCNodes()
+  let node
+
+  for (let index = 0; index < creatorNodes.length; index++) {
+    node = creatorNodes[index]
+    try {
+      const resp = await axios({
+        method: 'get',
+        url: `${node.endpoint}/file_lookup`,
+        params: {
+          filePath,
+          timestamp,
+          delegateWallet,
+          signature
+        },
+        responseType: 'stream',
+        timeout: 1000
+      })
+      if (resp.data) {
+        await writeStreamToFileSystem(resp.data, filePath)
+
+        // verify that the file written matches the hash expected if added to ipfs
+        const content = fs.createReadStream(filePath)
+        for await (const result of ipfsLatest.add(content, { onlyHash: true, timeout: 2000 })) {
+          if (cid !== result.cid.toString()) {
```

Comment:
hmmm how does this look if a given add operation results in >1 result value from IPFS? should we assert on the length being 0 too?

Reply:
is that even possible? also i copied this code from the saveFileForMultihash which checks the content this exact same way

Reply:
it is absolutely possible, the fact that it does not is due to the structure of our content not a property of IPFS

Reply:
i forgot the deets around this - base CID returned actually links to sub CIDs so i don't think we should assert here